.github/workflows/concrete_python_test_macos.yml

```diff
@@ -119,6 +119,31 @@ jobs:
           cp $GITHUB_WORKSPACE/frontends/concrete-python/Makefile .
           KEY_CACHE_DIRECTORY=./KeySetCache PYTEST_MARKERS="not dataflow and not graphviz" make pytest-macos
 
+      - name: Test notebooks
+        run: |
+          set -e
```

Comment:
I think you should create a new step before `Test` to prepare the test environment and avoid copy-pasting in both steps : 
```
- name: Prepare test environment
  run: |
           set -e
          export TEST_TMP_DIR=$(mktemp -d)
          echo "TEST_TMP_DIR=$TEST_TMP_DIR" >> "${GITHUB_ENV}"
          cd $TEST_TMP_DIR
          
          python3.10 -m venv .testenv
          . .testenv/bin/activate
          
          pip install $GITHUB_WORKSPACE/frontends/concrete-python/dist/*macos*.whl
          pip install -r $GITHUB_WORKSPACE/frontends/concrete-python/requirements.dev.txt
          # MacOS x86 have conflict between our OpenMP library, and one from torch
          # we fix it by using a single one (from torch)
          # see discussion: https://discuss.python.org/t/conflicting-binary-extensions-in-different-packages/25332/8
          
          find .testenv/lib/python3.10/site-packages -not \( -path .testenv/lib/python3.10/site-packages/concrete -prune \) -name 'lib*omp5.dylib' -or -name 'lib*omp.dylib' | xargs -n 1  ln -f -s $(pwd)/.testenv/lib/python3.10/site-packages/concrete/.dylibs/libomp.dylib
          cp -R $GITHUB_WORKSPACE/frontends/concrete-python/examples ./examples
          cp -R $GITHUB_WORKSPACE/frontends/concrete-python/tests ./tests
          
          cp $GITHUB_WORKSPACE/frontends/concrete-python/Makefile .
```

Then just reactivate on tests steps
```
      - name: Test ...
        run: |
          cd $TEST_TMP_DIR && . .testenv/bin/activate
          # run test ...
```

Reply:
ok sure
but then, do we agree that I can't have 

```
export TEST_TMP_DIR=$(mktemp -d)
```

anymore? I am not sure the $TEST_TMP_DIR exists after the step, so I would use a static name

Reply:
ok I've forced 

```
export TEST_TMP_DIR="testing_concrete_python"
```

you'll tell me if it's good for you

Reply:
the tmp dir should exists after the step, it's removed during the cleanup step (l 139)

Reply:
but both should work, but if the "testing_concrete_python" is corrupted by another job it may result in all other workflow failing, so I think mktemp is safer.

Reply:
ok let me check

Reply:
ok done, let me see if it works in the CI